installers/Config.hs

```diff
@@ -0,0 +1,123 @@
+{-# LANGUAGE LambdaCase          #-}
+{-# LANGUAGE NoImplicitPrelude   #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE RecordWildCards     #-}
+
+{-# LANGUAGE TypeOperators     #-}
+{-# LANGUAGE DataKinds         #-}
+module Config
+  ( generateConfig
+  , Request(..)
+  , OS(..), Cluster(..), Config(..)
+  , flag
+  , optReadLower, argReadLower
+  , Options(..), optionsParser
+  -- Re-export Turtle:
+  , options
+  ) where
+
+import Options.Generic (Generic, ParseRecord, type (<?>), unHelpful)
+
+import qualified Control.Exception
+
+import qualified Data.Char
+import qualified Data.ByteString
+import qualified Data.List                        as L
+import qualified Data.Map                         as Map
+import           Data.Maybe
+import           Data.Optional                       (Optional)
+import           Data.Semigroup
+import           Data.Text                           (Text, pack, unpack)
+import qualified Data.Text.Lazy                   as LT
+import qualified Data.Yaml
+
+import qualified Dhall
+import qualified Dhall.Import
+import qualified Dhall.JSON
+import qualified Dhall.Parser
+import qualified Dhall.TypeCheck
+
+import qualified GHC.IO.Encoding
+
+import qualified System.IO
+import qualified System.Exit
+
+import           Text.Printf                         (printf)
+import           Text.Trifecta.Delta                 (Delta(..))
+import           Turtle                              (optional)
+import           Turtle.Options
+
+import           Prelude
+import           Types
+
+
+
+diagReadCaseInsensitive :: (Bounded a, Enum a, Read a, Show a) => String -> Maybe a
```

Comment:
I don't know how to do this better but let's add an example comment here for motivation:

```
λ> data Foo = Bar | Baz deriving (Show, Enum, Bounded)
λ> let x = enumFromTo minBound maxBound :: [Foo]
λ> x
[Bar,Baz]
λ> fmap ((fmap toLower) . show) x
["bar","baz"]
```

Reply:
done